patroni/ha.py

```diff
@@ -884,6 +884,11 @@ def _run_cycle(self):
                 if msg is not None:
                     return msg
 
+            # is the data directory empty and we are the leader?
+            if self.state_handler.data_directory_empty() and self.has_lock():
```

Comment:
Why not to move this block no the line 889? It will eliminate double call of `self.state_handler.data_directory_empty()`
Everything else looks good to me.